### PR TITLE
feat: support client redirect

### PR DESCRIPTION
**Pull request by yixiaojiu**
### Description

It's a simple implentment that write additional HTML pages to static site that redirect the user to existing pages with JavaScript.

Refrence [docusaurus plugin-client-redirects](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-client-redirects)

### Linked Issues

https://github.com/YunYouJun/valaxy/issues/237

### Copilot Descriptions

copilot:all

### CODE REVIEW

**Comment by YunYouJun** [comment]
Thanks for your contribution! I need some time to review it, but if you need it, you can publish it yourself.

File: packages/valaxy-addon-client-redirects/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "valaxy-addon-client-redirects",
+  "version": "0.0.1",
+  "description": "Client Redirect for Valaxy",
+  "repository": "https://github.com/YunYouJun/valaxy/tree/main/packages/valaxy-addon-client-redirects",
+  "keywords": [
```
Comment (YunYouJun) [inline packages/valaxy-addon-client-redirects/package.json]: You can add your name in authors.

**Comment by YunYouJun** [comment]
This may not seem like a big feature, but it can be useful in some cases.

How do you feel about making it directly a built-in feature of valaxy?
Or is addon more suitable for it?

Let's discuss it?

**Comment by yixiaojiu** [comment]
> This may not seem like a big feature, but it can be useful in some cases.
> 
> How do you feel about making it directly a built-in feature of valaxy?
> Or is addon more suitable for it?
> 
> Let's discuss it?
> 
> 

You are right. Maybe building in valaxy is a good idea. I will move it later.

**Comment by yixiaojiu** [comment]
how about define the config in `site.config.ts`, something like this:

```ts
export default defineSiteConfig({
      redirects: [
        {
          from: '/foo',
          to: '/about',
        },
      ],
})
```

**Comment by YunYouJun** [comment]
It's OK.

If you want to know about some implementation issues in the code, feel free to ask me.

**Comment by YunYouJun** [comment]
If you are ready, you can change the title of this issue and @ me review.

**Comment by yixiaojiu** [comment]
@YunYouJun I have finished it.

**Comment by YunYouJun** [comment]
Thanks!

I have a few other ideas, such as adding e2e tests, redirection at the vue router level, and redirect configuration in frontmatter.

Anyway, let's get it up and running.

**Comment by YunYouJun** [comment]
image

We can even add this feature in the client side by `vue-router` `beforeEach`.

**Comment by yixiaojiu** [comment]
Ok, I will have a try later. (＾▽＾)

**Pull request merged**